tests/ui/traits/assoc-type-hrtb-normalization-30472.rs

```diff
@@ -0,0 +1,28 @@
+//@ check-pass
```

Comment:
Suggestion: can you add a brief doc comment on what this test is trying to exercise, and also add a link to #30472 in the doc comment?